draft-ietf-jose-json-web-proof.md

```diff
@@ -126,64 +126,88 @@ While `issue` and `confirm` only occur when a JWP is initially created by the is
+
+Payloads MAY contain Parameters, or an application might describe their own formats and layouts of payloads.
+
+
+## Proof
+
+A Proof consists of one or more octet string values generated from and processed by the selected algorithm.  Internally, the proof value may contain one or more cryptographic statements that are used to check the integrity protection of the header(s) and of all payloads.  Each of these statements may be a ZKP or a traditional cryptographic signature.  The algorithm is responsible for how these statements are serialized and ordered into a proof.
+
+# Parameter Requirements
+
+## Parameter Labeling Requirements
+
+Labels are the mechanism for semantically distinguishing parameters. It is important to describe the mechanism to reduce the risk
 of conflicts.
 
-There are three strategies for labeling header parameters:
+There are four strategies for labeling Parameters:
 
 1. Registered parameter labels. These labels are coordinated through
-the IANA "JSON Web Proof Header Parameters" registry, which protects
+the IANA "JSON Web Proof Parameters" registry, which protects
 against parameters having the same label.
 
-2. Collision-resistant parameter labels. These labels are not
+2. Separately registered claims. For compatibility, registered claims from the JSON Web Token Claims registry (#IANA.JWT.Claims] MAY be used as parameters in JSON serialized JWPs, and registered claims from the CBOR Web Token Claims registry (#IANA.CWT.Claims) may be used in CBOR serialized JWPs. The labels MUST indicate the claims came from an external registry, as described below.
+
+3. Collision-resistant parameter labels. These labels are not
 coordinated through IANA, but are otherwise namespaced to prevent
 conflict. One example would a string label representing the URI of a
 controlled resource, such as the HTTPS-hosted documentation of the
-header parameter.
+Parameter.
 
-3. Private parameter labels. These labels are not coordinated through
+4. Private parameter labels. These labels are not coordinated through
 IANA or another party, but are expected to only be used for testing
 or in closed environments.
 
-These classes of Header Parameters are intentionally parallel to those in Section 4 of [@RFC7515].
+### Use of the JSON Web Token Claims Registry
 
-## Registered Header Parameter Labels {#RegisteredHeaderParameterLabels}
+To use a claim from the JWT registry, the claim name is prefixed with a colon (":") to form the Parameter label. A colon prefix for parameter labels MUST NOT be used otherwise. Claims from the JWT registry SHOULD NOT be used when an appropriate JWP Parameter is available.
```

Comment:
I've thought about this since the colon prefix was first proposed a few days ago and I can't live with changing the claim names to add a prefix.  It will cause developer confusion and coding mistakes forever.